pkg/skaffold/deploy/helm.go

```diff
@@ -47,12 +52,16 @@ func NewHelmDeployer(cfg *v1alpha2.DeployConfig, kubeContext string, namespace s
 }
 
 func (h *HelmDeployer) Deploy(ctx context.Context, out io.Writer, builds []build.Build) error {
+	deployResults := []DeployResult{}
 	for _, r := range h.HelmDeploy.Releases {
-		if err := h.deployRelease(out, r, builds); err != nil {
+		results, err := h.deployRelease(out, r, builds)
+		if err != nil {
 			releaseName, _ := evaluateReleaseName(r.Name)
 			return errors.Wrapf(err, "deploying %s", releaseName)
 		}
+		deployResults = append(deployResults, results...)
 	}
+	labelDeployResults(deployResults)
```

Comment:
To avoid having to run this in each deployer, I would suggest that we change the deploy interface back to

` Deploy(ctx context.Context, out io.Writer, builds []build.Build) (DeployResult, error)`

That maybe looks like
```golang
type DeployResult struct {
  deployedArtifacts []DeployedArtifact
}
  
type DeployedArtifact struct {
  GVK string
  obj *runtime.Obj
}
``` 

The high level flow would be

```golang
func run() {
 buildResults := builder.build()
 deployResults := deployer.deploy(buildResults)
 label(cfg, deployResults)
}
```

Reply:
I had played around with having this live in the runner, but decided to go with putting it in each deployer for two reasons:
1) the runner will need to know less about what happened in its individually managed steps, and
2) labeling should be coupled to deploying, since we can't label unless we've successfully deployed.

That said, I can see arguments for moving it to the runner as well. If we add more deployers that's more boilerplate we have to remember to add, and the runner has a lot more information about the overall skaffold run which could be useful labeling information. Since the runner has all the skaffold config info, I think that's a strong enough argument for moving it back, so I'll go ahead with it and see how it looks.